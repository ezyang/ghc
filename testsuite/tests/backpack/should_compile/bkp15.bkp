unit p where
    signature T where
        data T
    signature H where
        import T
        f :: T -> T
unit timpl where
    module TImpl where
        data T = T
unit q where
    include timpl
    signature T(T) where
        import TImpl
    include p
    module A where
        import H
        import TImpl
        x = f T
unit r where
    include timpl
    module H where
        import TImpl
        f T = T
    module T(T) where
        import TImpl
    include q
