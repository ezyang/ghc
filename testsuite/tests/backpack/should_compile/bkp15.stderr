[1 of 5] Processing p
  [1 of 2] Compiling T[sig]           ( p/T.hsig, nothing )
  [2 of 2] Compiling H[sig]           ( p/H.hsig, nothing )
[2 of 5] Processing timpl
  Instantiating timpl
  [1 of 1] Compiling TImpl            ( timpl/TImpl.hs, bkp15-out/timpl/TImpl.o )
[3 of 5] Processing q
  [1 of 3] Compiling T[sig]           ( q/T.hsig, nothing )
  [2 of 3] Compiling H[sig]           ( q/H.hsig, nothing )
  [3 of 3] Compiling A                ( q/A.hs, nothing )
[4 of 5] Processing r-impl
  Instantiating r-impl
  [1 of 1] Including timpl
  [1 of 2] Compiling H                ( r-impl/H.hs, bkp15-out/r-impl/H.o )
  [2 of 2] Compiling T                ( r-impl/T.hs, bkp15-out/r-impl/T.o )
[5 of 5] Processing r
  Instantiating r
  [1 of 2] Including r-impl
  [2 of 2] Including q
    Instantiating q(r-impl:T, r-impl:H)
    [1 of 2] Including p
      Instantiating p(r-impl:T, r-impl:H)
      [1 of 2] Compiling T[sig]           ( p/T.hsig, nothing )
      [2 of 2] Compiling H[sig]           ( p/H.hsig, nothing )
    [2 of 2] Including timpl
    [1 of 3] Compiling T[sig]           ( q/T.hsig, nothing )
    [2 of 3] Compiling H[sig]           ( q/H.hsig, nothing )
    [3 of 3] Compiling A                ( q/A.hs, bkp15-out/q/q-5X8ly0nqp7UFKgRhoP1UWK/A.o )
